pkg/controller/volume/persistentvolume/index_test.go

```diff
@@ -1033,8 +1057,8 @@ func TestVolumeModeCheck(t *testing.T) {
 	blockMode := v1.PersistentVolumeBlock
 	filesystemMode := v1.PersistentVolumeFilesystem
 
-	// If feature gate is enabled, VolumeMode will always be defaulted
-	// If feature gate is disabled, VolumeMode is dropped by API and ignored
+	// If feature gate is enabled on apiserver, VolumeMode will always be defaulted
+	// If feature gate is disabled on apiserver, VolumeMode is dropped by API and ignored
```

Comment:
is/was this comment accurate? When does defaulting occur, is it only on object `create`? Even if the feature gates are correctly set for all components (kubelet, controller-manager, apiserver) isn't it possible for a nil VolumeMode to exist?

Reply:
@msau42 is it intentional that we fail volume operations like bind/mount/ if a nil VolumeMode is found (via CheckVolumeModeFilesystem which returns an error) when feature gate is enabled to somehow alert an admin that there is a misconfiguration?

Reply:
I can't find the conversation anymore, but I remember when discussing the alpha implementation with api-machinery, we confirmed that when we read the object from the api server, it will default the value as long as the feature gate is on, even if it wasn't stored with a value set.  So as long as all the feature gates are properly set, we should not encounter a nil field.

The main argument for relying on defaulting is that the defaulting behavior is explicit per api version.  If we handled a nil field case by defaulting it to some value, that behavior is not bound by any api contract.

Reply:
cc @liggitt if this makes sense

Reply:
fields set in defaulting, are returned from the API in read paths. (the read path for an API object goes `etcd -> default  -> convert to internal -> convert to requested version -> serialize -> send to requester`).

Reply:
OK, I've dropped the latest commit, imo let's keep the assumptions that field will always be populated and keep the weird test changes to account for it